### PR TITLE
Fix of problem in deletefile

### LINKED ISSUE #264: 52 new test failures on Hobart
Just checked in on CDash and as of the noon auto-test today, all of the tests fail on Hobart (local linux cluster with the Nag compiler). Yesterday we had one test fail. Today, we are getting runtime crashes with the following text:

pio_decomp_tests_2d:69817 terminated with signal 11 at PC=4523ec SP=7fff94d9b1b0.  Backtrace:

pio_decomp_tests_2d:69818 terminated with signal 11 at PC=4523ec SP=7ffebd714d50.  Backtrace:
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d(PIOc_deletefile+0xfc)[0x4523ec]
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d(piolib_mod_MP_pio_deletefile+0x102)[0x43e436]
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d(PIOc_deletefile+0xfc)[0x4523ec]
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d(nc_write_read_2d_col_decomp_pio_int_integer___+0x1f9c)[0x4262f8]
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d(pio_tf_test_driver__+0x6f)[0x431811]
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d(main+0x194)[0x4329b9]
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d(piolib_mod_MP_pio_deletefile+0x102)[0x43e436]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x7f20928bfb15]
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d(nc_write_read_2d_col_decomp_pio_int_integer___+0x1f9c)[0x4262f8]
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d[0x41bb21]
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d(pio_tf_test_driver__+0x6f)[0x431811]
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d(main+0x194)[0x4329b9]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x7f236fba4b15]
/scratch/cluster/katec/dashboard/build-Linux-3.10.0-327.el7.x86_64-x86_64-Nag-6.1-gcc-4.8.5-Nightly/tests/general/pio_decomp_tests_2d[0x41bb21]

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 69818 RUNNING AT h011
=   EXIT CODE: 1
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================

### PR DESCRIPTION
**Pull request by edhartnett**
Fixes #264.

### CODE REVIEW

File: tests/unit/test_nc4.c [hunk 1]
```diff
@@ -57,6 +57,56 @@ int dim_len[NDIM] = {NC_UNLIMITED, X_DIM_LEN, Y_DIM_LEN};
...
+        if ((ret = PIOc_closefile(ncid)))
+            ERR(ret);
+
+        /* Now delete the file. */
+        printf("%d Deleting %s...\n", my_rank, filename);
+        if ((ret = PIOc_deletefile(iosysid, filename)))
+            ERR(ret);
+
+        /* Make sure it is gone. */
+        /* if ((ret = PIOc_openfile(iosysid, &ncid, &(flavor[fmt]), filename, */
```
Comment (jedwards4b) [inline tests/unit/test_nc4.c]: Why is this commented out?

File: tests/unit/test_nc4.c [hunk 2]
```diff
@@ -278,6 +321,23 @@ int test_nc4(int iosysid, int num_flavors, int *flavor, int my_rank)
...
+int test_all(int iosysid, int num_flavors, int *flavor, int my_rank)
+{
+    int ret; /* Return code. */
+    
+    /* Test file deletes. */
+    if ((ret = test_deletefile(iosysid, num_flavors, flavor, my_rank)))
+        return ret;
+
+    /* /\* Test netCDF-4 functions. *\/ */
+    /* if ((ret = test_nc4(iosysid, num_flavors, flavor, my_rank))) */
```
Comment (jedwards4b) [inline tests/unit/test_nc4.c]: Why is this commented out?

File: tests/unit/test_nc4.c [hunk 3]
```diff
@@ -421,8 +476,8 @@ int main(int argc, char **argv)
             return ret;
 
         /* Run tests with async. */
-        if ((ret = test_async(my_rank, num_flavors, flavor, test_comm)))
-            return ret;
+        /* if ((ret = test_async(my_rank, num_flavors, flavor, test_comm))) */
```
Comment (jedwards4b) [inline tests/unit/test_nc4.c]: and finally - same question?

Comment (edhartnett) [inline tests/unit/test_nc4.c]: There is a problem with the error handling, in non-async, and it is not yet enabled in async.

In non-async it does not heed my settings for the IOsystem, and instead just mpi_aborts when I try to reopen the deleted file.

In async mode, setting the error handler does not yet work. I am working on that in another PR now.

So I want to merge this PR, which does fix the deletefile problem. Then I will merge another PR with the error handler issues address, and these parts of the test turned back on.

File: tests/unit/test_nc4.c (hunk 2 above)
Comment (edhartnett) [inline tests/unit/test_nc4.c]: This should not have been commented out and I've fixed that in the branch.

**Pull request merged**